azext_iot/iothub/edge_device_config.py

```diff
@@ -0,0 +1,469 @@
+# coding=utf-8
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+"""This module defines common values and functions for processing edge device configurations"""
+
+from typing import Optional, List, Dict
+from azext_iot.common.fileops import write_content_to_file
+from azext_iot.common.certops import create_root_certificate, load_ca_cert_info
+from azext_iot.common.shared import (
+    ConfigType,
+    DeviceAuthType,
+    NestedEdgeConfig,
+    NestedEdgeDeviceConfig,
+    EdgeContainerAuth,
+)
+from azext_iot.common.utility import process_json_arg, process_toml_content
+
+from azure.cli.core.azclierror import (
+    CLIInternalError,
+    FileOperationError,
+    InvalidArgumentValueError,
+    RequiredArgumentMissingError,
+)
+
+from azext_iot.operations.hub import _process_config_content
+from azext_iot.sdk.iothub.service.models import ConfigurationContent
+from knack.log import get_logger
+
+logger = get_logger(__name__)
+
+# Edge device TOML default values
+DEVICE_CONFIG_TOML = {
+    "hostname": "",
+    "provisioning": {
+        "device_id": "",
+        "iothub_hostname": "",
+        "source": "manual",
+        "authentication": {
+            "device_id_pk": "",
+            "method": "sas",
+            "trust_bundle_cert": "",
+        },
+    },
+    "edge_ca": {"cert": "file:///", "pk": "file:///"},
+    "agent": {"config": {"image": ""}, "name": "edgeAgent", "type": "docker"},
+    "connect": {
+        "management_uri": "unix:///var/run/iotedge/mgmt.sock",
+        "workload_uri": "unix:///var/run/iotedge/workload.sock",
+    },
+    "listen": {
+        "management_uri": "fd://aziot-edged.mgmt.socket",
+        "workload_uri": "fd://aziot-edged.workload.socket",
+    },
+    "moby_runtime": {"network": "azure-iot-edge", "uri": "unix:///var/run/docker.sock"},
+}
+
+EDGE_ROOT_CERTIFICATE_FILENAME = "iotedge_config_cli_root.pem"
+
+EDGE_DEVICE_BUNDLE_DEFAULT_FOLDER_NAME = "device_bundles"
+
+EDGE_CONFIG_SCRIPT_HEADERS = """
+# This script will attempt to configure a pre-installed iotedge as a nested node.
+# It must be run as sudo, and will modify the ca
+
+device_id="{}"
+cp config.toml /etc/aziot/config.toml
+"""
+EDGE_CONFIG_SCRIPT_HOSTNAME = """
+# ======================= Set Hostname =======================================
+
+read -p "Enter the hostname to use: " hostname
+if [ -z "$hostname" ]
+then
+    echo "Invalid hostname $hostname"
+    exit 1
+fi
+
+sed -i "s/{{HOSTNAME}}/$hostname/" /etc/aziot/config.toml
+"""
+EDGE_CONFIG_SCRIPT_PARENT_HOSTNAME = """
+# ======================= Set Parent Hostname =======================================
+
+read -p "Enter the parent hostname to use: " parent_hostname
```

Comment:
What about ensuring a device or its parent hostname is known? As in are `EDGE_CONFIG_SCRIPT_PARENT_HOSTNAME` and `EDGE_CONFIG_SCRIPT_HOSTNAME` must haves?